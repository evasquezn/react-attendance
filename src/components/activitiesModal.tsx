import React, { ReactElement, useState } from 'react';
import { createStyles, Theme, withStyles, WithStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import MuiDialogContent from '@material-ui/core/DialogContent';
import MuiDialogActions from '@material-ui/core/DialogActions';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Typography from '@material-ui/core/Typography';
import ActivityList from './activityList';
//#region Styling 

const styles = (theme: Theme) =>
  createStyles({
    root: {
      margin: 0,
      padding: theme.spacing(2),
    },
    closeButton: {
      position: 'absolute',
      right: theme.spacing(1),
      top: theme.spacing(1),
      color: theme.palette.grey[500],
    },
  });

const DialogContent = withStyles((theme: Theme) => ({
    root: {
      padding: theme.spacing(2),
    },
  }))(MuiDialogContent);
  
  const DialogActions = withStyles((theme: Theme) => ({
    root: {
      margin: 0,
      padding: theme.spacing(1),
    },
  }))(MuiDialogActions);

//#endregion

export interface IActivitiesModalProps {
    show: boolean,
    studentName: string,
    course:string,
    date: string,
};

export const ActivitiesModal = (props: IActivitiesModalProps): ReactElement => {
    const { show, studentName, course, date} = props;
    const [open,setOpen] = useState(show);
    const handleClose = ()=>{
        setOpen(false);
    };

    return (
        <div>
          <Dialog onClose={handleClose} aria-labelledby="customized-dialog-title" open={open}>
            <MuiDialogTitle disableTypography>
            <Typography variant="h6">Activity Log</Typography>          
                    <IconButton aria-label="close" onClick={handleClose}>
                    <CloseIcon />
                    </IconButton>
            </MuiDialogTitle>
            <DialogContent dividers>
             <ActivityList list={[]}/>  
            </DialogContent>
            <DialogActions>
              <Button autoFocus onClick={handleClose} color="primary">
                Save changes
              </Button>
            </DialogActions>
          </Dialog>
        </div>
      );
};

export default ActivitiesModal;