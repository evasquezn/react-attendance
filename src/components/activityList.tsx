import React,{ ReactElement } from 'react';
import { createStyles, Theme, makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem, { ListItemProps } from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import InboxIcon from '@material-ui/icons/Inbox';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: '100%',
      maxWidth: 360,
      backgroundColor: theme.palette.background.paper,
    },
  }),
);

interface IList {
    activity: string,
    totalTime: string
};

export interface IActivityListProps {
    list: Array<IList>,
};

const ActivityList=(props: IActivityListProps): ReactElement => {
  const classes = useStyles();
  const { list } = props;
  return (
    <div className={classes.root}>
      <List component="nav" aria-label="main mailbox folders">
        <ListItem button>
          <ListItemIcon>
            <InboxIcon />
          </ListItemIcon>
          <ListItemText primary="Inbox" />
        </ListItem>
      </List>
    </div>
  );
}
export default ActivityList;