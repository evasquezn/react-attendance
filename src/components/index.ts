export * from "./studentAttendanceReport";
export * from "./reportHeader";
export * from "./reportGrid";
export * from "./progressSpinner";
export * from "./reportFooter"
