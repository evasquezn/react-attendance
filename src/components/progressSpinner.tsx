import React, { ReactElement } from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
import Backdrop from '@material-ui/core/Backdrop';

//#region Styling 

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    backdrop: {
      zIndex: theme.zIndex.drawer + 1,
      color: '#fff',
    },
  }),
);

//#endregion

export interface IProgressSpinnerProps {
    show: boolean
};

export const ProgressSpinner = (props: IProgressSpinnerProps): ReactElement => {
    const classes = useStyles();
    const { show } = props;

    return (
        <Backdrop className={classes.backdrop} open={show}>
            <CircularProgress color="inherit" />
        </Backdrop>
    );
};

export default ProgressSpinner;