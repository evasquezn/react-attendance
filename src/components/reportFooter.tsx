import React from 'react';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import { makeStyles } from '@material-ui/core/styles';
import { useReportContext } from ".";

// Define our styles for material-ui
const useStyles = makeStyles({
  root: {
    height: "80px",
    maxHeight: "80px",
    width: "100%",

    backgroundColor: "green"
  },
  footerWidgetContainer: {
    height: "46px",
    padding:"3px 6px"
  },
  footerWidgetLabelText: {
    fontSize: 14,
    lineHeight: "64px",
    fontWeight: "bold",
    justifySelf: "bottom",
    verticalAlign: "bottom",
  }
});

export const ReportFooter = function(props: any) {
  const classes = useStyles();
  const { reportContext, setReportContext } = useReportContext();
  const { page, pageSize } = reportContext;

  return (
    <Container className={classes.root} maxWidth="lg">
        <Box display="flex" flexDirection="row" flexWrap="nowrap" width="100%">
            {/* TODO: create footer paging and alphabet filter */}
        </Box>
    </Container>
  );
}
export default ReportFooter;