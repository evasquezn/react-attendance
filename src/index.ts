import { StudentAttendanceReportRenderer } from "renderers"
import "@fontsource/roboto"; 

export { StudentAttendanceReportRenderer as StudentAttendanceReport };

