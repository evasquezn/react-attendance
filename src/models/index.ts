export * from "./attendanceLogEntry";
export * from "./courseData";
export * from "./courseActivityEntry";
export * from "./student";
export * from "./studentData";
export * from "./studentAttendanceReportProps";
export * from "./reportContext";
export * from "./timeEntry";
export * from "./activity";
