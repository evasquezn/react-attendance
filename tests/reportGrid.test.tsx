import React from 'react';
import renderer from 'react-test-renderer';
import { ReportGrid } from "../src/components";
import { IStudent } from "../src/models";


test('Grid Renders Correctly', () => {
  const selectedDate = new Date("2021/04/04");
  const studentData: IStudent[] = [];
  const attendanceReportAPIBaseUrl = "https://this.dummy.com/api"
  const onError = function() { };
  const beforeLoad = function() { };
  const afterLoad = function() { };

  const component = renderer.create(
    <ReportGrid
      apiUrl={attendanceReportAPIBaseUrl}
      selectedDate={selectedDate}
      studentData={studentData}
      onError={onError}
      beforeLoad={beforeLoad}
      afterLoad={afterLoad}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});