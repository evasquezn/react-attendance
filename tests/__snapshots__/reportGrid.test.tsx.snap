// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Renders Correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div />
    <div
      className="MuiTableContainer-root"
    >
      <table
        aria-label="student attendance log"
        aria-labelledby="tableTitle"
        className="MuiTable-root makeStyles-table-3"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={20}
            >
              <button
                aria-label="expand row"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                    />
                  </svg>
                </span>
              </button>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={30}
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-8 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    aria-label="select all students"
                    checked={false}
                    className="PrivateSwitchBase-input-11"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
              scope="col"
              width={140}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Student
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                 
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
              scope="col"
              width="auto"
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                 
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={70}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Sun
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                04/04
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={70}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Mon
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                04/05
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={70}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Tue
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                04/06
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={70}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Wed
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                04/07
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={70}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Thu
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                04/08
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={70}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Fri
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                04/09
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={70}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Sat
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                04/10
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-sizeSmall"
              scope="col"
              width={110}
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Total Time
              </h6>
              <span
                className="MuiTypography-root MuiTypography-caption"
              >
                (hh:mm:ss)
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        />
      </table>
    </div>
  </div>
</div>
`;
